content/zh-tw/blue-green-deployment.md

```diff
@@ -0,0 +1,30 @@
+---
+title: 藍綠部署
+status: Completed
+category: 概念
+tags: ["方法論", "應用程式", ""]
+---
+
+## 是什麼 {#what-it-is}
+
+藍綠部署是一種以最小的停機時間更新執行中的電腦系統的方法。
+維運商維護兩個環境，被稱為 "藍" 和 "綠"。
+一個提供正式服務的流量（所有使用者目前正在使用的版本），另一個則是需要升級的服務。
+一旦非活躍（綠色）環境中的測試結束，
+正式服務流量會被切換過來（通常會使用[負載平衡器](/zh-tw/load-balancer/)）。
+注意，藍綠部署通常意思是要切換整個環境，包括許多[服務](/zh-tw/service/)。
+令人困惑的是，有時這個術語被用於一個系統內的單個服務。
+為了避免這個歧異，提到單個元件時，最好使用 "零停機部署" 一詞。
+
+## 解決的問題 {#problem-it-addresses}
+
+在更新那些缺乏向後相容性而必須"同步"改變的軟體時，藍綠部署允許最短的停機時間。
+例如，藍綠部署適用於一個由網站和資料庫組成的線上商店，
+該商店需要更新，但新版本的資料庫不能與就版本的網站一起使用，反之亦然。
+在這樣的情況下，兩者需要同時改變。
+如果在正式環境這樣做，客戶會注意到停機時間。
+
+## 如何幫助 {#how-it-helps}
+
+對於需要以最小的停機時間進行更新的非雲端原生軟體來說，藍綠部署是一種適合的方法。
+然而，它的使用通常是一種 "感覺"，傳統軟體需要重新設計，以便可以單獨更新元件。
```

Comment:
我認為這邊的 "smell" 不是 "感覺" 的意思，而是 "臭味" 或是 "警訊" 的意思。我會建議這麼翻：

```
然而，當我們需要用到藍綠部署的時候，通常這會是一種 "警訊" ，意味著這塊遺留軟體需要重新設計，以便可以單獨更新它的元件。
```

原文:

> However, its use is normally a “smell” that legacy software needs to be re-engineered so that components can be updated individually.